harness/determined/common/experimental/determined.py

```diff
@@ -483,12 +483,27 @@ def remove_oauth_client(self, client_id: str) -> None:
     def stream_trials_metrics(
         self, trial_ids: List[int], group: str
     ) -> Iterable[metrics.TrialMetrics]:
-        """Get a stream of metrics for this trial.
+        warnings.warn(
+            "Determined.stream_training_metrics is deprecated."
+            "Use Determined.iter_trials_metrics instead",
+            FutureWarning,
+            stacklevel=2,
+        )
+        return self.iter_trials_metrics(trial_ids=trial_ids, group=group)
+
+    def iter_trials_metrics(
+        self, trial_ids: List[int], group: str
+    ) -> Iterable[metrics.TrialMetrics]:
+        """Generate an iterator of metrics for the passed trials.
+
+        This function opens up a persistent connection to the Determined master to receive trial
+        metrics. For as long as the connection remains open, the generator it returns yields the
+        TrialMetrics it receives.
 
         Arguments:
-            trial_ids: The trial IDs to stream metrics for.
-            group: The metric group to stream.  Common values are "validation" and "training", but
-                group can be any value passed to master when reporting metrics during training
+            trial_ids: The trial IDs to iterate over metrics for.
+            group: The metric group to iterate over.  Common values are "validation" and "training",
+                but group can be any value passed to master when reporting metrics during training
```

Comment:
to avoid ending in a preposition:

> Arguments:
    trial_ids: The list of trial IDs whose metrics will be iterated over.
    group: The metric group that Determined will iterate over. Common values include "validation" and "training". 
           However, the metric group can be any value reported to the master during training, typically 
           through a context's `report_metrics`.